### PR TITLE
Add config for the backup dir path

### PR DESCRIPTION
**Pull request by tobiasreischmann**
This PR allows to set a path for the backup dir. This way it is easier for system administrators to place the backups folder in a separate storage space.

### CODE REVIEW

File: settings.php [hunk 1]
```diff
@@ -38,6 +38,11 @@
         get_string('config_delay_duration_desc', 'tool_lifecycle'),
         183 * 24 * 60 * 60)); // Dafault value is 180 days.
 
+    $settings->add(new admin_setting_configtext('tool_lifecycle/backup_path',
```
Comment (Dagefoerde) [inline settings.php]: This could be an `admin_setting_configdirectory` instead, which provides some sort of feedback as to whether the directory exists.

Comment (Dagefoerde) [reply settings.php]: ... in that context, have a look at how it is used in https://github.com/moodle/moodle/blob/8cf4c97217230e74e57e41f24e8afbbc8393eb79/admin/settings/analytics.php#L134-L142, which includes auto-creation of the default path.

Comment (tobiasreischmann) [reply settings.php]: I am against autocreation. But the type is changed.

Comment (tobiasreischmann) [reply settings.php]: Also found the old reference and fixed it.

**Comment by Dagefoerde** [comment]
:shipit:

**Pull request merged**